snapcraft/tests/test_parser.py

```diff
@@ -66,6 +71,15 @@ def tearDown(self):
         except FileNotFoundError:
             pass
 
+    def setUp(self):
+        super().setUp()
+        tempdir = tempfile.mkdtemp()
+        patcher = mock.patch('snapcraft.internal.parser._get_base_dir')
+        base_dir = patcher.start()
+        base_dir.return_value = tempdir
```

Comment:
I've converted the parser tests to use fixtures.TempDir().  I'm not 100% certain I'm handling the cleanUp correctly so please review that part carefully.